### PR TITLE
fix(endorsement-system-api): Updated models, post and put requests in endorsements systems

### PR DESCRIPTION
**Pull request by alexdiljar**
# ...

Opened Date and admin Lock added to Endorsement List model and show name added to endorsements.

## What

General petitions can be locked by admin, admin lock functionality on get requests to be implemented. Closed and opened dates added to endorsement lists to control when general petitions should close. Party applications get the same functionality.

## Why

To be able to implement general petitions

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by rafnarnason — APPROVED** [review]
LGTM - just one comment in the review

File: libs/application/template-api-modules/src/lib/modules/templates/party-letter/party-letter.service.ts [hunk 1]
```diff
@@ -25,6 +25,8 @@ const CREATE_ENDORSEMENT_LIST_QUERY = `
     }
   }
 `
+// TODO: change this date when new election
+const defaultClosedDate = new Date('2021-09-15T00:00:00.000Z')
```
Comment (saevarma) [inline libs/application/template-api-modules/src/lib/modules/templates/party-letter/party-letter.service.ts:29]: Shouldn't this be in an env variable?

File: libs/application/templates/party-application/src/hooks/useIsEndorsementClosed.ts [hunk 2]
```diff
@@ -18,6 +18,6 @@ export const useIsClosed = (endorsementListId: string) => {
       },
     },
   )
-
+  // TODO: change to check if closed date is bigger thann date now
```
Comment (saevarma) [inline libs/application/templates/party-application/src/hooks/useIsEndorsementClosed.ts:21]: Why not adding this check now?

File: libs/application/templates/party-letter/src/hooks/useIsEndorsementClosed.ts [hunk 3]
```diff
@@ -18,6 +18,6 @@ export const useIsClosed = (endorsementListId: string) => {
       },
     },
   )
-
+  // TODO: change to check if date now inbetween opened and closed date
```
Comment (saevarma) [inline libs/application/templates/party-letter/src/hooks/useIsEndorsementClosed.ts:21]: Why not adding this check now?

File: libs/application/template-api-modules/src/lib/environments/environment.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export default {
+  defaultclosedDate: new Date(
```
Comment (baering) [inline libs/application/template-api-modules/src/lib/environments/environment.ts:2]: `defaultClosedDate`?

File: libs/application/template-api-modules/src/lib/environments/environment.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+    process.env.PARTY_ENDORSEMENTLISTS_DEFAULT_CLOSED_DATE ||
+      '2021-09-15T00:00:00.000Z',
+  ),
+}
```
Comment (eirikurn) [inline libs/application/template-api-modules/src/lib/environments/environment.ts:6]: Not sure if we want an environments file here for template specific configuration. Think we should move this to the template api folder for now.

**Review by saevarma — APPROVED** [review]
LGTM

**Pull request merged**